0035-feature-flags/README.md

```diff
@@ -0,0 +1,418 @@
+and modified at runtime from ctx for maximal flexibility. (TODO Storage and retrieval implications)
+- Centralize implementation within the CLI to provide a clear implementation guidance on how to declare feature flags.
+- TODO Discuss global fallback to ctx in API
+- Automate docs.zarf.dev page with 1:1 mapping to flags as they're declared and documented in code.
+
+### User Stories (Optional)
+
+- As a Zarf maintainer I want to add a new feature, get feedback, make revisions over time before fully releasing and 
+maintaining it.
+- As a Zarf maintainer I want to deprecate features with clarity on when they will be removed as well as trialing removal
+by disabling the feature before it's fully removed.
+
+
+- As a Zarf user I want to get involved in new experimental features, and opt-in to provide alpha feedback or beta testing.
+- As a Zarf user I want to continue using Zarf without my critical workflows getting disrupted by experimental features.
+- As a Zarf user I want deeper clarity if a feature that I use will be deprecated.
+
+### FIXME Risks and Mitigations
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Zarf ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+-->
+
+- One critical pitfall to avoid with feature flags is creating _unexpected_ behavior, both for end users and maintainers.
+  Unclear state about which flags are enabled and not, and with and without defaults. We have some solutions below under
+  risks and mitigations on how to account for this.
+- Developer experience (DX) and user experience (UX) are both critical feature flag adoption.
+- User feedback is key.
+- Rollout considerations, like can we _backport_ flags to previous features so that they can be associated with verions
+or disabled? 
+
+## FIXME Design Details
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss that.
+-->
+
+### Types
+Below is the proposed data model for Feature Flags, including the `Flag` type and supporting fields in the `flag` pkg.
+```
+pkg flag 
+...
+
+type Name string
+type Description string
+
+# TODO Validate Since matches semver
+type Since string
+
+type Stage string
+var (
+  StageAlpha Stage = "alpha"
+  StageBeta Stage = "beta"
+  StageGA Stage = "ga"
+  StageDeprecated Stage = "deprecated"
+)
+
+type Flag struct {
+  # Name stores the name of the feature flag.
+  Name
+  # Description describes how the flag is used.
+  Description
+  # Enabled is set if a feature is set.
+  Enabled bool
+  # Default is set if a feature is enabled by default, without being set by users.
+  Default bool
```

Comment:
I'd suggest not to store Enabled next to Default, but rather keep this structure minimal such that it will only hold information about feature gate, such as: name, description, version, enabled, stage. Separately, you'll want to keep a registry of currently enabled flags (those with enabled=true out of the box and those enabled by users explicitly through flags), and devs should always use the registry helper to know if it's on or off. 

This separation ensures we don't accidentally enable something that should not be enabled.